astropy/units/format/generic.py

```diff
@@ -58,41 +57,50 @@ class Generic(Base):
 
     _show_scale = True
 
-    def __init__(self):
-        # Build this on the class, so it only gets generated once.
-        if '_parser' not in Generic.__dict__:
-            Generic._parser, Generic._lexer = self._make_parser()
+    _tokens = (
+        'DOUBLE_STAR',
+        'STAR',
+        'PERIOD',
+        'SOLIDUS',
+        'CARET',
+        'OPEN_PAREN',
+        'CLOSE_PAREN',
+        'FUNCNAME',
+        'UNIT',
+        'SIGN',
+        'UINT',
+        'UFLOAT'
+    )
+
+    @classproperty(lazy=True)
+    def _all_units(cls):
+        return cls._generate_unit_names()
+
+    @classproperty(lazy=True)
```

Comment:
These ones don't really have to be lazy at all, right?  I'd think only `_all_units` would need to be calculated only once.